onnxruntime/test/util/include/asserts.h

```diff
@@ -4,7 +4,7 @@
 #pragma once
 
 #include "core/common/status.h"
-#include "core/session/onnxruntime_c_api.h"
+#include "core/session/onnxruntime_cxx_api.h"
```

Comment:
I believe this is causing the following error: `libonnxruntime_providers_cuda_ut.so: undefined symbol: OrtGetApiBase`

The test/providers/cuda/test_cases/reduction_functions_test.cc file (part of onnxruntime_providers_cuda_ut) [includes this asserts.h file](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/test/providers/cuda/test_cases/reduction_functions_test.cc#L13). Now that onnxruntime_cxx_api.h is included transitively, it needs to link with onnxruntime_session because the C++ api header calls [OrtBaseApi()->GetApi() to initialize the global api_ variable.](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_cxx_api.h#L118)

Reply:
Thanks! Split out the assert headers to avoid this dependency as the new usage for the OrtStatus asserts is much simpler so I think it's worth maintaining.